client/lib/olark/index.js

```diff
@@ -151,6 +151,12 @@ const olark = {
 		}
 	},
 
+	fetchUserEligibility() {
```

Comment:
Calling `configureOlark` more than once in the page lifetime will cause unpredictable issues because we'll end up listening to [these](https://github.com/Automattic/wp-calypso/blob/fix/1827-update-contact-option/client/lib/olark/index.js#L118) events more than once.

I think what you should do here is create a new function based on [this](https://github.com/Automattic/wp-calypso/blob/fix/1827-update-contact-option/client/lib/olark/index.js#L173) method that uses the Olark Api to change the configuration.

Your new method could look something like this:
```
	fetchUserEligibility() {
		this.getOlarkConfiguration()
			.then( ( configuration ) => this.updateOlarkGroupAndEligibility( configuration ) )
			.catch( ( error ) => this.handleError( error ) );
	},
	updateOlarkGroupAndEligibility( wpcomOlarkConfig = {} ) {
		const isUserEligible = ( 'undefined' === typeof wpcomOlarkConfig.isUserEligible ) ? true : wpcomOlarkConfig.isUserEligible;
		olarkApi( 'api.chat.setOperatorGroup', { group: wpcomOlarkConfig.group } );
		olarkActions.setUserEligibility( isUserEligible );
	}
```
See [api.chat.setOperatorGroup](https://www.olark.com/api#api.chat.setOperatorGroup)